test/Makefile

```diff
@@ -38,4 +45,16 @@ test_cheap:
 	cd sharness && make
 	cd 3nodetest && make
 
-.PHONY: all clean
+test_race:
+	cd sharness && make GOFLAGS=-race TEST_EXPENSIVE=1
+	cd 3nodetest && make GOFLAGS=-race
+	cd dependencies && make GOFLAGS=-race
+
+IPFS-BUILD-OPTIONS: FORCE
+	@VARS='$(GOFLAGS)'; \
+	    if test x"$$VARS" != x"`cat $@ 2>/dev/null`" ; then \
+		echo "*** new Go flags ***"; \
+		echo "$$VARS" >$@; \
+            fi
```

Comment:
I copied what is done in the Git Makefile for these kind of things.
It is just checking if the content of the GOFLAGS variable is different from what is in the IPFS-BUILD-OPTIONS file, and if it is different, it puts the content of the GOFLAGS variable into the file.

We could have something like:

```
bin/program-normal
bin/program-race
```

and then a `bin/program` symlink that points to one or the other of the above binaries. But I am not sure it would be simpler and this way we couldn't have concurent tests going on.

To have concurent tests going on, we would need to modify the test scripts before launching them with something like sed 's/ipfs/ipfs-race/g' but it could be tricky to get right too.

Or maybe I am missing something?